cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractContext.java

```diff
@@ -169,7 +169,7 @@ public DiscoveryServiceClient getDiscoveryServiceClient() {
     return discoveryServiceClient;
   }
 
-  protected static Map<String, String> getMetricsContext(Program program, String runId) {
+  public static Map<String, String> getMetricsContext(Program program, String runId) {
```

Comment:
True - already started doing that, but reverted so that we have incremental changes. Will be done in next iterations